### PR TITLE
[DYN-2755] Preview disabled nodes can be renamed with double click

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

[DYN-2755](https://jira.autodesk.com/browse/DYN-2755)
This adds the ability to the preview disabled nodes to interpret the double click event and open rename dialog box.
UI Automation Test PR: [#24](https://git.autodesk.com/Dynamo/DynamoUIAutomation/pull/24)

Before:
preview-before

After:
preview-after2

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs [hunk 1]
```diff
@@ -376,6 +376,15 @@ private void OnPreviewMouseLeftButtonDown(object sender, MouseButtonEventArgs e)
...
             BringToFront();
+            if (e.ClickCount == 2)
+            {
+                Debug.WriteLine("Name double clicked!");
+                if (ViewModel != null && ViewModel.RenameCommand.CanExecute(null))
+                {
+                    ViewModel.RenameCommand.Execute(null);
+                }
+                e.Handled = true;
+            }
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: The part is not clear to me is that for nodes in other state, how are we handling the double click? On a different route?

Comment (mmisol) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: Should Python script nodes with preview disabled also be editable? This code only accounts for rename, but the code editor of those nodes can be opened as well by double clicking. It actually depends on the part of the node where you double click.

Comment (zeusongit) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: Just confirmed with Sol, nodes should/can be renamed in any state.

Comment (mmisol) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: But isn't this change overriding the behaviour of the handler, by handling the double click as a rename and marking the event as handled? Can you check that you can still edit a Python node by double clicking after your change?

Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: Yup, @zeusongit if you can confirm the other cases, I am fine with this change

Comment (zeusongit) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:387]: Can confirm.
python preview

**Pull request merged**